### ISSUE TITLE
[5.0] SelfHandling command without input not working.

### ISSUE THREAD

**Issue by fredriklaason** [laravel/framework#6946]
Let's say I have SomeCommand that implements SelfHandling and shouldn't handle any input.

In controller with DispatchesCommands trait

    $this->dispatch(SomeCommand::class);

does not work.

In controller with DispatchesCommands trait

    $this->dispatchFromArray(SomeCommand::class, []);

does work.

using `dispatch()` from the DispatchesCommands trait generates 

    ErrorException in Dispatcher.php line 296:
    get_class() expects parameter 1 to be object, string given

In other words SomeCommand is never instantiated.

**Comment by ibrasho** [comment]
I think you should try `$this->dispatch(new SomeCommand);`.

**Comment by RomainLanz** [comment]
According to your error, you didn't pass an instance of your class but the name.
`SomeCommand::class` return 'SomeCommand' and not an instance, try what @ibrasho tell. It should work.

**Comment by fredriklaason** [comment]
Just thought laravel magic would handle something like that :)

**Comment by ibrasho** [comment]
Even Laravel's magic is limited. ;-)
